### PR TITLE
fix win

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?

Fixes # https://github.com/openvinotoolkit/openvino/issues/26844

force deleting openvino model and garbage collector during model export to prevent locking files in tmp dir due to mmap.
Additionally fixed installation from source on windows (in setup.py, subprocess.run raises FileNotFound error if can not find git) and refactoring weights compression part moving model size calculation during export and read model only if weights compression should be applied (based on model size or explicitly specified configuration, prevents OOM and invalid IR errors if nncf not installed)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

File: optimum/intel/openvino/modeling_base.py [hunk 1]
```diff
@@ -737,3 +737,9 @@ def _incompatible_inputs_warning(self, inputs: Dict):
                 return f"Got unexpected inputs: `{input_name}` set to {type(inputs[input_name])} while expected to be {dtype}."
 
         return None
+
+    def __del__(self):
+        if hasattr(self, "request"):
+            del self.request
+        if hasattr(self, "model"):
+            del self.model
```
Comment (nikita-savelyevv) [inline optimum/intel/openvino/modeling_base.py:745]: Is this required or just to be safe?

File: optimum/intel/openvino/modeling_decoder.py [hunk 2]
```diff
@@ -404,6 +404,14 @@ def _make_stateful(self):
         patch_stateful(self.config, self.model)
         self.stateful = True
 
+    def __del__(self):
+        if hasattr(self, "request"):
+            del self.request
+        if hasattr(self, "_original_model"):
+            del self._original_model
+        if hasattr(self, "model"):
+            del self.model
```
Comment (nikita-savelyevv) [inline optimum/intel/openvino/modeling_decoder.py:413]: Same question

**Review by nikita-savelyevv — APPROVED** [review]
:+1:

File: optimum/intel/openvino/modeling_base.py (hunk 1 above)
Comment (eaidova) [reply optimum/intel/openvino/modeling_base.py:745]: I added for experiment to check that process exit issues will be resolved if we remove model and request in destructor. checking that attribute exists added for safety (if somebody already remove them before)

**Comment by eaidova** [comment]
I still getting segmentation fault (probably caused double free memory corruption in my experiments with resolving tmp dir destruction, so I moved changes required for compression into separated PR to unblock people that waiting this issue resolution https://github.com/huggingface/optimum-intel/pull/925/

**Pull request closed without merging**